ModelDescriber/EnumModelDescriber.php

```diff
@@ -17,6 +17,12 @@ public function describe(Model $model, Schema $schema)
             $enums[] = $enumCase->value;
         }
 
+        $reflectionEnum = new \ReflectionEnum($enumClass);
+        if ($reflectionEnum->isBacked() && 'int' === $reflectionEnum->getBackingType()->getName()) {
+            $schema->type = 'integer';
+        } else {
+            $schema->type = 'string';
+        }
         $schema->type = is_subclass_of($enumClass, \IntBackedEnum::class) ? 'int' : 'string';
```

Comment:
I'm not really a fan of the if-else combined with the multi places where `$schema->type` gets set.

```suggestion
        $schema->type = is_subclass_of($enumClass, \IntBackedEnum::class) || ($reflectionEnum->isBacked() && 'int' === $reflectionEnum->getBackingType()->getName())
            ? 'integer' 
            : 'string';
```

Reply:
Adil Ishaq